tests/system/providers/amazon/aws/example_ec2.py

```diff
@@ -150,6 +153,22 @@ def parse_response(instance_ids: list):
     )
     # [END howto_operator_ec2_terminate_instance]
     terminate_instance.trigger_rule = TriggerRule.ALL_DONE
+
+    # [START howto_operator_ec2_hibernate_instance]
+    hibernate_instance = EC2HibernateInstanceOperator(
+        task_id="hibernate_instace",
+        instance_id=instance_id,
+    )
+    # [END howto_operator_ec2_hibernate_instance]
+    hibernate_instance.trigger_rule = TriggerRule.ALL_DONE
+    
+    # [START howto_operator_ec2_reboot_instance]
+    reboot_instance = EC2RebootInstanceOperator(
+        task_id="reboot_instace",
+        instance_id=instance_id,
+    )
+    # [END howto_operator_ec2_reboot_instance]
+    reboot_instance.trigger_rule = TriggerRule.ALL_DONE
```

Comment:
I dont think this sequence would work. At that point the instance is terminated, hence you cannot hibernate it. I would also reboot it before hibernating it. I would follow this order:
- reboot
- hibernate
- terminate

Reply:
The order is actually dictated by the `chain` below. The hibernate/reboot are happening before the terminate there. But I would still change the order of the tasks here to reflect the order of the chain just for completeness.

Reply:
Nice catch!

Reply:
[Updated the order to align with the chain order](https://github.com/apache/airflow/pull/35770/files#diff-c2d4bfb7a226fbdae32c03ef2b0382f74ea4875c12c32370677069b165ca8c37R184)